concrete/src/Validator/ClosureValidator.php

```diff
@@ -1,56 +1,71 @@
 <?php
+
 namespace Concrete\Core\Validator;
 
+use ArrayAccess;
+use Closure;
+
+/**
+ * A generic validator cabable of describing itself and validating mixed values using closures.
+ */
 class ClosureValidator implements ValidatorInterface
 {
-    /** @var \Closure */
+    /**
+     * The closure that handles validation.
+     *
+     * @var \Closure
+     */
     protected $validator_closure;
 
-    /** @var \Closure */
+    /**
+     * The closure that returns requirements.
+     *
+     *@var \Closure
+     */
     protected $requirements_closure;
 
     /**
      * ClosureValidator constructor.
      *
-     * @param \Closure $validator_closure function(ClosureValidator $validator,
-     *                                             mixed $passed,
-     *                                             \Concrete\Core\Error $error = null): bool
+     * @param \Closure $validator_closure function(ClosureValidator $validator, mixed $passed, \Concrete\Core\Error $error = null): bool
      * @param \Closure $requirements_closure function(ClosureValidator $validator): array
+     *
+     * @example
      */
-    public function __construct(\Closure $validator_closure, \Closure $requirements_closure)
+    public function __construct(Closure $validator_closure, Closure $requirements_closure)
     {
         $this->validator_closure = $validator_closure;
         $this->requirements_closure = $requirements_closure;
     }
 
     /**
-     * Set the closure that handls validation
-     * function(ClosureValidator $validator, mixed $passed, \Concrete\Core\Error $error = null): bool.
+     * Set the closure that handles validation.
      *
      * @param \Closure $validator_closure
+     *
+     * @example function(ClosureValidator $validator, mixed $passed, \Concrete\Core\Error $error = null): bool
```

Comment:
`@example` is meant to describe the current structural element, this comment is a bit confusing because it makes me think that this is an example for how I'm supposed to use this method. Instead just provide the function signature with the `@param` doc line.